### PR TITLE
fix: disable the confirm button when there is a blocking alert

### LINKED ISSUE #27147: [Bug]: Misleading copy in the existing friction modal of alert system
### Describe the bug

We're displaying some misleading copy in the existing friction modal of alert system. 
Separating the copy update of the friction modal from this issue so we can target the copy change for v12.3. The other remaining fixes for alert system can be done in follow up releases.

### Expected behavior

Display a more generic and toned down message on the friction modal.

### Screenshots/Recordings

Screenshot 2024-09-13 at 15 06 31

https://github.com/user-attachments/assets/00d759ec-1fde-4f46-8534-6719f5264374

### Steps to reproduce

1. Go to test dapp
2. Don't have funds
3. Trigger malicious mint erc20 on Sepolia
4. Click on Review alerts
or
1. Go to test dapp
2. Click Sign In With Ethereum Bad Domain
3. Click Confirm
4. See the friction modal 

### Error messages or log output

_No response_

### Detection stage

During release testing

### Version

12.3.0

### Build type

None

### Browser

Brave

### Operating system

MacOS

### Hardware wallet

_No response_

### Additional context

_No response_

### Severity

_No response_

### PR DESCRIPTION
**Pull request by vinistevam**
## **Description**
This PR aims to disable the confirm button when there is a blocking alert.

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/27347?quickstart=1)

## **Related issues**

Fixes: https://github.com/MetaMask/metamask-extension/issues/27147

## **Manual testing steps**

1. Go to test dapp
2. Don't have funds
3. Trigger malicious mint erc20 on Sepolia
4. Click on Review alerts
or
1. Go to test dapp
2. Click Sign In With Ethereum Bad Domain
3. Click Confirm
4. See the friction modal 

## **Screenshots/Recordings**

[Screencast from 23-09-2024 17:57:22.webm](https://github.com/user-attachments/assets/4c00284c-93d3-4b1b-9553-39eda67b7924)

## **Pre-merge author checklist**

- [x] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Comment by sleepytanya** [comment]
Malicious Mint ERC20 -` Review alerts` button is disabled:


https://github.com/user-attachments/assets/3475ea0a-8371-438c-aa72-ea52d05d7d53


Sign in with Ethereum (bad domain) - `Review alerts` and `Confirm` buttons are active:



https://github.com/user-attachments/assets/7a23ce00-8bfe-4573-b53f-1639b417cfec



I think the below is related to the dvevlopment builds being slow but just in case:
- 'Cancel' has to be clicked 20 times before it works
- sometimes blue 'Be careful' blocker banner is not displayed

Screenshot 2024-09-23 at 19 45 03



https://github.com/user-attachments/assets/7afe7a6d-8aea-4789-b987-e3539c51a2ac

**Comment by sleepytanya** [comment]
Account has enough funds, `Confirm` enabled as expected:

Screenshot 2024-09-23 at 22 07 24

**Review by OGPoyraz — APPROVED** [review]

File: ui/pages/confirmations/components/confirm/footer/footer.tsx [hunk 1]
```diff
@@ -86,12 +107,16 @@ const ConfirmButton = ({
           block
           danger
           data-testid="confirm-footer-button"
-          disabled={hasUnconfirmedDangerAlerts ? false : disabled}
+          disabled={getButtonDisabledState(
+            hasUnconfirmedDangerAlerts,
+            hasDangerBlockingAlerts,
+            disabled,
+          )}
```
Comment (jpuri) [inline ui/pages/confirmations/components/confirm/footer/footer.tsx:114]: Above condition looks bit unclear to me if `hasDangerBlockingAlerts` is true disabled will always be false.

**Review by pedronfigueiredo — APPROVED** [review]

File: ui/pages/confirmations/components/confirm/footer/footer.tsx (hunk 1 above)
Comment (vinistevam) [reply ui/pages/confirmations/components/confirm/footer/footer.tsx:114]: If the `hasDangerBlockingAlerts` disabled will always be true, so the button will be disabled until the user resolves the blocking alert in the current case this is the insufficient balance alert.

**Pull request merged**